chm_utils/clients/snowflake.py

```diff
@@ -0,0 +1,101 @@
+ARE_DEPS_AVAILABLE = True
+try:
```

Comment:
it's not so common to see conditional importing. why is this necessary here?

Reply:
if it is like optional depedencies, one idea is to handle it in pyproject.tml file
ex
[project.optional-dependencies]

Reply:
agreed, but i'd like to be able to call `from chm_utils.clients import Snowflake` even when dependencies aren't available (see dependency groups). therefore, we throw an importerror only when the client is initialised.